engine/src/main/java/org/terasology/config/flexible/settings/StringSetting.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible.settings;
+
+import org.terasology.config.flexible.validators.SettingValueValidator;
+import org.terasology.engine.SimpleUri;
+
+public class StringSetting extends SettingImpl<String> {
```

Comment:
I see you made String, Float, Integer and Double settings. I would suggest to add BooleanSetting and perhaps LongSetting. Byte, Char and Short would complete the list but I feel they are probably a bit overkill for our needs. I wouldn't say no if you added them though... :wink: